framework/src/utils/PetscDMMoose.C

```diff
@@ -631,7 +632,7 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
               PenetrationInfo * pinfo = lit->second;
               std::map<dof_id_type, std::vector<dof_id_type> > & node_to_elem_map =
               dmm->_nl->_fe_problem.mesh().nodeToElemMap();
-              if (pinfo && pinfo->isCaptured())
+              if (pinfo && (pinfo->isCaptured() || dmm->_include_all_contact_nodes))
```

Comment:
> DMMoose should really be agnostic to anything about contact. You're using a "leaky abstraction" here. Anything contact related should be only in the ContactSplit.

I noticed this before. The code has been written like this way by the previous developer. Options for both the contact problems and the general ones are all put together.  We do not declare DMMoose as a class.  I think we should be going to do that. But this will be a big refactor. Possibly let us merge this one? And then we could decide if try to refactor the code and extract the code for contact problems.